compiler/semantic/scope.go

```diff
@@ -65,6 +77,15 @@ func (s *Scope) DefineFunc(f *dag.Func) error {
 	return nil
 }
 
+func (s *Scope) DefineOp(o *ast.OpDecl) error {
+	b := s.tos()
+	if _, ok := b.symbols[o.Name]; ok {
+		return fmt.Errorf("symbol %q redefined", o.Name)
+	}
+	b.Define(o.Name, o)
+	return nil
+}
```

Comment:
This is so similar to DefineAs that it doesn't seem worth its weight. Maybe just change the type of the second DefineAs parameter to any and use that?

Reply:
Yeah that's a fair point.